application/utxohandler/utxohandler.go

```diff
@@ -760,15 +760,17 @@ func (ut *UTXOHandler) addOneFastSync(txn *badger.Txn, utxo *objs.TXOut) error {
 			utils.DebugTrace(ut.logger, err)
 			return err
 		}
-		value, err := utxo.Value()
-		if err != nil {
-			utils.DebugTrace(ut.logger, err)
-			return err
-		}
-		err = ut.valueIndex.Add(txn, utxoID, owner, value)
-		if err != nil {
-			utils.DebugTrace(ut.logger, err)
-			return err
+		if !utxo.IsDeposit() {
+			value, err := utxo.Value()
+			if err != nil {
+				utils.DebugTrace(ut.logger, err)
+				return err
+			}
+			err = ut.valueIndex.Add(txn, utxoID, owner, value)
+			if err != nil {
+				utils.DebugTrace(ut.logger, err)
+				return err
+			}
```

Comment:
Prefer early break over nesting for guard checks:

```suggestion
		if !utxo.IsDeposit() {
		  break
		}
  	value, err := utxo.Value()
  	if err != nil {
  		utils.DebugTrace(ut.logger, err)
  		return err
  	}
  	if err := ut.valueIndex.Add(txn, utxoID, owner, value); err != nil {
	  	utils.DebugTrace(ut.logger, err)
	  	return err
```